### PR TITLE
Option to reduce generated PDF file size

### PR DESCRIPTION
**Pull request by dvwright**
Feature request. The PDF's being generated for me are often multi page and average ~ 42MB, this optional parameter if set, reduces them to ~ 1.4MB, without much loss of quality in my case. This PR implements my desired results. Thanks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ domToPdf(element, options, function() {
 * `excludeClassNames` - array of strings, list of class names of elements to exclude from PDF, e.g. `['Loading', 'ExcludeMeFromPdf']`
 * `overrideWidth` - number, overrides a width of a container DOM element 
 * `proxyUrl` - string, e.g. `/api/proxyImage?url=`, a route in your app which renders images on your domain in order to avoid problems with CORS with the images on a DOM
+* `reduceSize` - boolean, e.g. `true/false`, if set, the PDF created is MUCH smaller (up to 40x smaller) quality may be reduced as well. (the default is false)
```
Comment (omazin) [inline README.md]: I'd prefer having the `compression` parameter
compression | string | compression of the generated JPEG, can have the values 'NONE', 'FAST', 'MEDIUM' and 'SLOW
-- | -- | --

File: index.js [hunk 2]
```diff
@@ -223,7 +224,11 @@ downloadPdf = (dom, options, cb) => {
         pdf.addPage();
       }
       imgData = pageCanvas.toDataURL('image/PNG');
-      pdf.addImage(imgData, 'PNG', 0, 0, a4Width, pageHeight);
+      if (reduceSize) {
```
Comment (omazin) [inline index.js]: with the `compression` parameter we get rid of the `if/else` and will have just
```javascript
pdf.addImage(imgData, 'PNG', 0, 0, a4Width, pageHeight, undefined, compression);
```

File: index.js [hunk 3]
```diff
@@ -84,8 +84,9 @@ downloadPdf = (dom, options, cb) => {
   let overlayCSS;
   let pageHeightPx;
   let proxyUrl;
+  let reduceSize;
```
Comment (omazin) [inline index.js]:
```javascript
let compression = 'NONE';
```

**Review by omazin — CHANGES_REQUESTED** [review]

**Comment by ovvn** [comment]
Hi @dvwright, thank you for the PR, please consider changes in the review.

**Comment by dvwright** [comment]
Hi @ovvn great suggestions, implemented, thanks

**Review by omazin — APPROVED** [review]

**Pull request merged**